src/test/kotlin/uk/gov/justice/hmpps/offenderevents/e2e/HMPPSDomainEventsIntTest.kt

```diff
@@ -431,50 +398,21 @@ class HMPPSDomainEventsTest : QueueListenerIntegrationTest() {
           """
             .trimIndent(),
         )
-        // send control message so we can check messages have been processed
-        sendToTopic(
-          "OFFENDER_BOOKING-REASSIGNED",
-          // language=JSON
-          """
-            {
-              "eventType":"OFFENDER_BOOKING-REASSIGNED",
-              "bookingId":"1201234",
-              "eventDatetime":"2021-02-08T14:41:11.526762Z",
-              "offenderIdDisplay":"A9999CA",
-              "nomisEventType":"OFF_BKB_UPD",
-              "offenderId":"2620073",
-              "previousOffenderIdDisplay":"A5694DR",
-              "previousOffenderId":"5260560"
-            }
-            """
-            .trimIndent(),
-        )
-        await().until { getNumberOfMessagesCurrentlyOnHMPPSEventTestQueue() == 2 }
+        await().until { getNumberOfMessagesCurrentlyOnHMPPSEventTestQueue() == 1 }
       }
 
       @Test
-      fun `will publish a merge event (along with a control message)`() {
-        val hmppsEventMessages = geMessagesCurrentlyOnHMPPSTestQueue()
-        assertThat(hmppsEventMessages).hasSize(1)
-        val mergeDomainEvent = hmppsEventMessages.first()
-
-        assertThatJson(mergeDomainEvent).node("eventType").isEqualTo("prison-offender-events.prisoner.merged")
-        assertThatJson(mergeDomainEvent).node("occurredAt").asString()
-          .satisfies(
-            ThrowingConsumer { dateTime: String? ->
-              assertThat(dateTime).isEqualTo("2022-11-02T00:39:05.070936Z")
-            },
-          )
-        assertThatJson(mergeDomainEvent).node("publishedAt").asString()
-          .satisfies(
-            ThrowingConsumer { dateTime: String? ->
-              assertThat(OffsetDateTime.parse(dateTime))
-                .isCloseTo(OffsetDateTime.now(), within(10, ChronoUnit.SECONDS))
-            },
-          )
-        assertThatJson(mergeDomainEvent).node("additionalInformation.reason").isEqualTo("MERGE")
-        assertThatJson(mergeDomainEvent).node("additionalInformation.removedNomsNumber").isEqualTo("A5694DR")
-        assertThatJson(mergeDomainEvent).node("additionalInformation.bookingId").asString().isEqualTo("1201234")
+      fun `will publish a merge event`() {
+        val mergeDomainEvent = geMessagesCurrentlyOnHMPPSTestQueue().first()
+
```

Comment:
simply json processing by switching to the spring library and a couple of extensions below